common/network-common/src/test/java/org/apache/spark/network/client/TransportClientFactorySuite.java

```diff
@@ -224,4 +226,23 @@ public void closeFactoryBeforeCreateClient() throws IOException, InterruptedExce
     factory.close();
     factory.createClient(TestUtils.getLocalHost(), server1.getPort());
   }
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Test
+  public void fastFailConnectionInTimeWindow() throws IOException, InterruptedException {
+    TransportClientFactory factory = context.createClientFactory();
+    TransportServer server = context.createServer();
+    int unreachablePort = server.getPort();
+    server.close();
+    try {
+      factory.createClient(TestUtils.getLocalHost(), unreachablePort, true);
+    } catch (Exception e) {
+      assert(e instanceof IOException);
+    }
+    expectedException.expect(IOException.class);
+    expectedException.expectMessage("fail this connection directly");
+    factory.createClient(TestUtils.getLocalHost(), unreachablePort, true);
```

Comment:
shall we reset the `expectedException` after this test case completes?

Reply:
and AFAIK junit has `assertThrows` to check exception and error message.

Reply:
It seems that we do not need reset the expectedException, it is uniq for each test.

Yes, junit5 provide assertThrows api, but our junit version is junit4.

Reply:
How do we associate `expectedException` to this specific test case?

Reply:
I think I get your point.
Do you mean that add `expectedException = ExpectedException.none();` after this test case complete?

I just refer the usage in SparkSubmitCommandBuilderSuite, it is only used before this pr.

Reply:
https://github.com/junit-team/junit4/blob/435d41f0d45cfdbc1a38e1ad4eb1d5300da533f9/src/main/java/org/junit/Rule.java#L27-L30

Here is the comments, it is just for each \@Test

Reply:
https://github.com/junit-team/junit4/blob/435d41f0d45cfdbc1a38e1ad4eb1d5300da533f9/src/main/java/org/junit/Rule.java#L27-L42

Here is an example, this folder is a RULER.
As described in the comments, it would create a temporary folder before
 each test method, and deletes it after each.
So, I think \@RULE should be unique for each test.

Reply:
yea, let's reset to `ExpectedException.none()`.

Reply:
done.